### PR TITLE
DEVEX-2307: add-project-new-external-upload-flag

### PR DESCRIPTION
**Pull request by PetratDNA**
DEVEX-2307
APPS-1462

* added support for externalUploadRestricted to DXProject.new(), .update()
* added it among recognized fields
* extended existing tests with this parameter

### CODE REVIEW

File: src/python/dxpy/utils/describe.py [hunk 1]
```diff
@@ -446,7 +447,9 @@ def print_project_desc(desc, verbose=False):
     if 'containsPHI' in desc:
         print_json_field('Contains PHI', desc['containsPHI'])
     if 'databaseUIViewOnly' in desc and desc['databaseUIViewOnly']:
-        print_json_field('Database UI View Only', desc['databaseUIViewOnly'])
+        print_json_field('External Upload Restricted', desc['databaseUIViewOnly'])
```
Comment (commandlinegirl) [inline src/python/dxpy/utils/describe.py:450]: What are we removing "Database UI View Only" here? Looks like a copy paste error.

File: src/python/dxpy/utils/describe.py [hunk 2]
```diff
@@ -446,7 +447,9 @@ def print_project_desc(desc, verbose=False):
...
+    if 'externalUploadRestricted' in desc and desc['externalUploadRestricted']:
+        print_json_field('E', desc['externalUploadRestricted'])
```
Comment (commandlinegirl) [inline src/python/dxpy/utils/describe.py:452]: Please update the key.

**Review by commandlinegirl — CHANGES_REQUESTED** [review]
Thanks for the contribution!

File: src/python/dxpy/utils/describe.py (hunk 1 above)
Comment (PetratDNA) [reply src/python/dxpy/utils/describe.py:450]: Yes, sorry. I missclicked and updated different line.

File: src/python/dxpy/utils/describe.py (hunk 2 above)
Comment (PetratDNA) [reply src/python/dxpy/utils/describe.py]: Updated.

**Review by commandlinegirl — APPROVED** [review]
LGTM, thank you!

**Pull request merged**